src/contracts/PurchaseBundler.ts

```diff
@@ -0,0 +1,101 @@
+import { decodeAbiParameters, encodeAbiParameters, Hex } from 'viem';
+
+import { Wallet } from '@/contracts';
+import { getContracts } from '@/deploys';
+import { purchaseBundlerAbi } from '@/generated/blockchain/v6';
+import { LoanToMslLoanType } from '@/utils/loan';
+
+import { BaseContract } from './BaseContract';
+
+export class PurchaseBundler extends BaseContract<typeof purchaseBundlerAbi> {
+  constructor({ walletClient }: { walletClient: Wallet }) {
+    const { PurchaseBundler } = getContracts(walletClient.chain);
+
+    super({
+      walletClient,
+      address: PurchaseBundler,
+      abi: purchaseBundlerAbi,
+    });
+  }
+
+  static SignableRepaymentData = [
+    {
+      name: 'data',
+      type: 'tuple',
+      components: [
+        { name: 'loanId', type: 'uint256' },
+        { name: 'callbackData', type: 'bytes' },
+        { name: 'shouldDelegate', type: 'bool' },
+      ],
+    },
+  ];
+
+  static LoanRepaymentData = [
```

Comment:
TODO: move this to MSLV6

Reply:
This should be read from abis. I'm pretty sure we can make it for this.